lib/state/terms_and_conditions.dart

```diff
@@ -101,17 +91,17 @@ class TermsAndConditionAcceptance extends Cubit<TermsAndConditionsAcceptanceStat
   void onChange(Change<TermsAndConditionsAcceptanceState> change) {
     super.onChange(change);
 
+    if (change.currentState == change.nextState) {
+      return;
+    }
     // TODO: Pass success/failure status to notification service.
-    _persistAcceptedVersionIfChanged(change.nextState.accepted?.version, change.currentState.accepted?.version);
+    _persistAcceptedVersionIfChanged(change.nextState.accepted);
   }
 
-  Future<void> _persistAcceptedVersionIfChanged(String? nextAcceptedVersion, String? currentAcceptedVersion) {
-    if (nextAcceptedVersion == currentAcceptedVersion) {
-      return Future.value();
-    }
+  Future<void> _persistAcceptedVersionIfChanged(AcceptedTermsAndConditions? nextAcceptedVersion) {
```

Comment:
The method name no longer reflects what this does. This always persists something.